server/images.go

```diff
@@ -768,43 +769,42 @@ func PruneDirectory(path string) error {
 	return nil
 }
 
-func PushModel(ctx context.Context, name string, regOpts *registryOptions, fn func(api.ProgressResponse)) error {
-	mp := ParseModelPath(name)
+func PushModel(ctx context.Context, name model.Name, opts registryOptions, fn func(api.ProgressResponse)) error {
 	fn(api.ProgressResponse{Status: "retrieving manifest"})
 
-	if mp.ProtocolScheme == "http" && !regOpts.Insecure {
-		return fmt.Errorf("insecure protocol http")
+	m, err := ParseNamedManifest(name)
+	if err != nil {
+		return err
 	}
 
-	manifest, _, err := GetManifest(mp)
+	scheme := "https"
```

Comment:
cc: @bmizerany 
I know you were making changes in this area recently, so just a heads up of the scheme changes here.

Reply:
All good. I had a little bridge patch. Landing the fix I had is still fair game, and it looks like this is a winner.